http/vertx/src/main/java/io/cloudevents/http/vertx/CeVertx.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2018 The CloudEvents Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents.http.vertx;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.CloudEventBuilder;
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpServerRequest;
+
+import java.net.URI;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static io.cloudevents.CloudEvent.CLOUD_EVENTS_VERSION_KEY;
+import static io.cloudevents.CloudEvent.EVENT_ID_KEY;
+import static io.cloudevents.CloudEvent.EVENT_TIME_KEY;
+import static io.cloudevents.CloudEvent.EVENT_TYPE_KEY;
+import static io.cloudevents.CloudEvent.EVENT_TYPE_VERSION_KEY;
+import static io.cloudevents.CloudEvent.HEADER_PREFIX;
+import static io.cloudevents.CloudEvent.SCHEMA_URL_KEY;
+import static io.cloudevents.CloudEvent.SOURCE_KEY;
+
+public final class CeVertx {
+
+    private CeVertx() {
+        // no-op
+    }
+
+    public static void writeToHttpClientRequest(final CloudEvent<?> ce, final HttpClientRequest request) {
+
+        request
+                // read required headers
+                .putHeader("content-type", "application/json")
+                .putHeader(CLOUD_EVENTS_VERSION_KEY, ce.getCloudEventsVersion())
```

Comment:
You can create optimised headers with `HttpHeaders.createOptimized("my-header")`, the optimized headers are `CharSequence` encoded to bytes directly.